packages/core/base/src/adapter.ts

```diff
@@ -137,10 +137,17 @@ export function scopePollingDetectionStrategy(detect: () => boolean): void {
         }
     }
 
-    // Strategy #1: Try detecting every second.
+    // Strategy #1: Try detecting every second for a fixed number of times.
+    let intervalAttempts = 5;
     const interval =
         // TODO: #334 Replace with idle callback strategy.
-        setInterval(detectAndDispose, 1000);
+        setInterval(function () {
+            detectAndDispose();
+            intervalAttempts--;
+            if (intervalAttempts <= 0) {
+                clearInterval(interval);
+            }
```

Comment:
The impetus behind this poller was to handle this use case (#327):

1. You arrive on site A that tells you to install a wallet.
2. You install it by whatever means.
3. Without refreshing site A, the newly installed wallet becomes available.

This change would cause a regression.

Hmm. It may be time to _actually_ complete #334, but to go _even_ further.

1. Schedule detection as an idle callback so as to be a good citizen.
2. _Pause_ detection when the window blurs/backgrounds, and resume it when the window focuses/foregrounds. Do this with a combination of `blur/focus` and `visibilitychange` events.

Reply:
U rite

Reply:
Reopened #334 and #337